aten/src/ATen/native/cuda/jit_utils.cpp

```diff
@@ -18,13 +18,21 @@
 #include <cstdlib>
 #include <string>
 
-#if BUILD_JITERATOR_WITH_CACHE
-  // Uses POSIX headers, which is why these are guarded behind BUILD_JITERATOR_WITH_CACHE
-  // TODO: C++17 has the fileystem header, which may replace these
+// TODO: C++17 has the fileystem header, which may replace these
+#ifdef _WIN32
+  #include <process.h>
+  #include <direct.h>
+  #include <io.h>
+  #define access _access
```

Comment:
can you add a comment here on why you are doing this defines? It's a clever trick to reuse existing code, but it'll be nice to describe it explicitly.

Reply:
Thanks. It's a fairly common code pattern (also used in third_party/tbb/examples/pipeline/square/gen_input.cpp)
Just added an explainer on top.